src/wayland/input_method_manager/mod.rs

```diff
@@ -0,0 +1,223 @@
+//! use smithay::{
+//!     delegate_seat, delegate_tablet_manager, delegate_input_method_manager,
+//!     delegate_text_input_manager,
+//! };
+//! use smithay::wayland::seat::{Seat, SeatState, SeatHandler, XkbConfig};
+//! use smithay::wayland::input_method_manager::{
+//!     InputMethodManagerState,
+//!     InputMethodSeatTrait
+//! };
+//! use smithay::wayland::text_input_manager::{
+//!     TextInputManagerState,
+//!     TextInputSeatTrait
+//! };
+//!
+//! # struct State { seat_state: SeatState<Self> };
+//!
+//! delegate_seat!(State);
+//! // Delegate input method handling for State to InputMethodManagerState.
+//! delegate_input_method_manager!(State);
+//! delegate_text_input_manager!(State);
+//!
+//! # let mut display = wayland_server::Display::<State>::new().unwrap();
+//! # let display_handle = display.handle();
+//!
+//! let seat_state = SeatState::<State>::new();
+//!
+//! // implement the required traits
+//! impl SeatHandler for State {
+//!     fn seat_state(&mut self) -> &mut SeatState<Self> {
+//!         &mut self.seat_state
+//!     }
+//! }
+//!
+//! // Add the seat state to your state and create manager globals
+//! InputMethodManagerState::new::<State>(&display_handle);
+//! TextInputManagerState::new::<State>(&display_handle);
+//!
+//! // create the seat
+//! let seat = Seat::<State>::new(
+//!     &display_handle,          // the display
+//!     "seat-0",                 // the name of the seat, will be advertized to clients
+//!     None                      // insert a logger here
+//! );
+//!
+//! seat.text_input();
+//! // Add text input capabilities to a seat, needed for the input method to work
+//! seat.add_input_method(XkbConfig::default(), 200, 25);
+//! // Add input method capabilities to a seat
+//!
+//! ```
+//! ### Run usage
+//!
+//! Once the input method and text input cabailities have been added to a seat,
+//! use the [`seat.input_method().set_point`] function to set the top left point
+//! of a focused surface. This is used to calculate the popup surface location.
+//!
+//! Then use the [`seat.input_method().coordinates`] and [`seat.input_method().with_surface`]
+//! functions to draw the popup surface.
+//!
+
+use wayland_server::{backend::GlobalId, Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New};
+
+use wayland_protocols_misc::zwp_input_method_v2::server::{
+    zwp_input_method_manager_v2::{self, ZwpInputMethodManagerV2},
+    zwp_input_method_v2::ZwpInputMethodV2,
+};
+
+use crate::wayland::seat::Seat;
+
+pub use input_method::{InputMethodHandle, InputMethodUserData};
+pub use input_method_keyboard_grab::InputMethodKeyboardUserData;
+pub use input_method_popup_surface::InputMethodPopupSurfaceUserData;
+
+use super::{seat::XkbConfig, text_input_manager::TextInputHandle};
+
+const MANAGER_VERSION: u32 = 1;
+
+mod input_method;
+mod input_method_keyboard_grab;
+mod input_method_popup_surface;
+
+/// Extends [Seat] with input method functionality
+pub trait InputMethodSeatTrait {
```

Comment:
Maybe just call it `InputMethodSeat`, the "Trait" is a little redundant.

Reply:
Fixed